engine/src/main/scala/cromwell/engine/workflow/lifecycle/EngineWorkflowExecutionActor.scala

```diff
@@ -0,0 +1,86 @@
+package cromwell.engine.workflow.lifecycle
+
+import akka.actor.{ActorRef, LoggingFSM, Props}
+import cromwell.engine.backend.BackendCallJobDescriptor
+import cromwell.engine.workflow.lifecycle.EngineWorkflowExecutionActor._
+import wdl4s.Call
+
+object EngineWorkflowExecutionActor {
+
+  /**
+    * States
+    */
+  sealed trait EngineWorkflowExecutionActorState { def terminal = false }
+  sealed trait EngineWorkflowExecutionActorTerminalState extends EngineWorkflowExecutionActorState { override def terminal = true }
+
+  case object WorkflowExecutionPendingState extends EngineWorkflowExecutionActorState
+  case object WorkflowExecutionInProgressState extends EngineWorkflowExecutionActorState
+  case object WorkflowExecutionSuccessfulState extends EngineWorkflowExecutionActorTerminalState
+  case object WorkflowExecutionFailedState extends EngineWorkflowExecutionActorTerminalState
+  case object WorkflowExecutionAbortedState extends EngineWorkflowExecutionActorTerminalState
+
+  /**
+    * State data
+    */
+  final case class EngineWorkflowExecutionActorData()
+
+  /**
+    * Commands
+    */
+  sealed trait EngineWorkflowExecutionActorCommand
+  case object StartExecutingWorkflowCommand extends EngineWorkflowExecutionActorCommand
+  case object RestartExecutingWorkflowCommand extends EngineWorkflowExecutionActorCommand
+  case object AbortExecutingWorkflowCommand extends EngineWorkflowExecutionActorCommand
+
+  /**
+    * Responses
+    */
+  sealed trait EngineWorkflowExecutionActorResponse
+  case object WorkflowExecutionSucceededResponse extends EngineWorkflowExecutionActorResponse
+  case object WorkflowExecutionAbortedResponse extends EngineWorkflowExecutionActorResponse
+  final case class WorkflowExecutionFailedResponse(reasons: Map[ActorRef, Throwable]) extends EngineWorkflowExecutionActorResponse
+
+  def props(workflowTag: String, backendAssignments: Map[Call, String]): Props = Props(EngineWorkflowExecutionActor(workflowTag, backendAssignments))
+}
+
+final case class EngineWorkflowExecutionActor(workflowTag: String, backendAssignments: Map[Call, String]) extends LoggingFSM[EngineWorkflowExecutionActorState, EngineWorkflowExecutionActorData] {
```

Comment:
So I was convinced that we weren't really Letting anything Crash, since we were accumulating errors and then passing a Message (just as an exception...) so for this PR I'll say "we use Messages" but it should be not too difficult to transition that in the future